internal/fourslash/tests/formatDocumentInPlace_test.go

```diff
@@ -0,0 +1,22 @@
+package fourslash_test
+
+import (
+	"testing"
+
+	"github.com/microsoft/typescript-go/internal/fourslash"
+	"github.com/microsoft/typescript-go/internal/testutil"
+)
+
+func TestFormatDocumentInPlace(t *testing.T) {
+	t.Parallel()
+	defer testutil.RecoverAndFail(t, "Panic on fourslash test")
+	const content = `/*1*/for (;;) { }
+/*2*/for (var x;x<0;x++) { }
+/*3*/for (var x ;x<0 ;x++) { }`
+	f := fourslash.NewFourslash(t, nil /*capabilities*/, content)
+	f.FormatDocument(t)
+	// After formatting, verify the formatted content
+	f.CurrentFileContentIs(t, `for (; ;) { }
+for (var x; x < 0; x++) { }
+for (var x; x < 0; x++) { } `)
```

Comment:
This extra space is undesirable. I've filed a bug at https://github.com/microsoft/typescript-go/issues/1997, but if it's easy to fix it would be nice.